indexing-service/src/test/java/org/apache/druid/indexing/seekablestream/SupervisorResourceConfigMergeTest.java

```diff
@@ -0,0 +1,533 @@
+import org.apache.druid.server.security.AuthConfig;
+import org.apache.druid.server.security.AuthenticationResult;
+import org.apache.druid.server.security.Authorizer;
+import org.apache.druid.server.security.AuthorizerMapper;
+import org.apache.druid.server.security.Resource;
+import org.apache.druid.server.security.ResourceAction;
+import org.apache.druid.server.security.ResourceType;
+import org.easymock.Capture;
+import org.easymock.EasyMock;
+import org.easymock.EasyMockRunner;
+import org.easymock.EasyMockSupport;
+import org.easymock.Mock;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.Response;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+@RunWith(EasyMockRunner.class)
+public class SupervisorResourceConfigMergeTest extends EasyMockSupport
+{
+  private static final ObjectMapper OBJECT_MAPPER = TestHelper.makeJsonMapper();
+  @Mock
+  private TaskMaster taskMaster;
+
+  @Mock
+  private SupervisorManager supervisorManager;
+
+  @Mock
+  private HttpServletRequest request;
+
+  @Mock
+  private AuthConfig authConfig;
+
+  @Mock
+  private AuditManager auditManager;
+
+  private SupervisorResource supervisorResource;
+
+  @Before
+  public void setUp()
+  {
+    supervisorResource = new SupervisorResource(
+        taskMaster,
+        new AuthorizerMapper(null)
+        {
+          @Override
+          public Authorizer getAuthorizer(String name)
+          {
+            return (authenticationResult, resource, action) -> {
+              if (authenticationResult.getIdentity().equals("druid")) {
+                return Access.OK;
+              } else {
+                if (resource.getType().equals(ResourceType.DATASOURCE)) {
+                  if (resource.getName().equals("datasource2")) {
+                    return Access.deny("not authorized.");
+                  } else {
+                    return Access.OK;
+                  }
+                } else if (resource.getType().equals(ResourceType.EXTERNAL)) {
+                  if (resource.getName().equals("test")) {
+                    return Access.deny("not authorized.");
+                  } else {
+                    return Access.OK;
+                  }
+                }
+                return Access.OK;
+              }
+            };
+          }
+        },
+        OBJECT_MAPPER,
+        authConfig,
+        auditManager
+    );
+  }
+
+  /**
+   * Tests that when a new spec is submitted without taskCountStart, the merge function
+   * follows the priority: provided taskCountStart > provided taskCount > existing taskCount > provided taskCountMin.
+   * In this test, existing spec has taskCount=5, new spec has no taskCount set,
+   * so merge should use existing taskCount (5).
+   */
```

Comment:
Can we remove the javadocs and update the method name to include the expectation/condition?